src/main/piece/tetris_piece.cpp

```diff
@@ -195,10 +182,43 @@ void TetrisPiece::shift( int direction_unit )
 
 void TetrisPiece::set_block_locations( std::vector<std::unique_ptr<Point>> param_block_locations )
 {
-  block_locations = move( param_block_locations );
+  current_block_locations = move( param_block_locations );
+}
+
+void TetrisPiece::set_next_block_locations( vector<unique_ptr<Point>> param_block_locations )
+{
+  next_block_locations = move( param_block_locations );
 } 
 
 void TetrisPiece::add_block_location( std::unique_ptr<Point> point )
 {
-  block_locations.push_back( move( point ) );
+  current_block_locations.push_back( move( point ) );
+}
+
+vector<unique_ptr<Point>> TetrisPiece::rotate_block_locations()
+{
+  int minimum_x = 0;
+  vector<unique_ptr<Point>> rotated_block_locations;
+  for( auto& point : current_block_locations )
+  {
+    int new_x = -1 * point -> get_y();
+    int new_y = point -> get_x();
```

Comment:
A linear rotation, ninety degrees about the origin, produced these x and y coordinates. They are the result of the product of the rotation matrix, [(0, -1), (1, 0)], with each point.